docs/Homebrew-homebrew-core-Merge-Checklist.md

```diff
@@ -1,22 +1,130 @@
+
+Depend on as little stuff as possible. Disable X11 functionality if possible.
+For example, we build Wireshark, but not the heavy GUI.
+
+Homebrew is about Unix software. Stuff that builds to an `.app` should
+be in Homebrew Cask instead.
+
+## Naming
+
+The name is the strictest item, because avoiding a later name change is
+desirable.
+
+Choose a name that’s the most common name for the project.
+For example, we initially chose `objective-caml` but we should have chosen `ocaml`.
+Choose what people say to each other when talking about the project.
+
+Formulae that are also packaged by other package managers (e.g. Debian, Ubuntu) should be
+named consistently (subject to minor differences due to Homebrew formula naming conventions).
+
+Add other names as aliases as symlinks in `Aliases` in the tap root. Ensure the
+name referenced on the homepage is one of these, as it may be different and have
+underscores and hyphens and so on.
+
+We now accept versioned formulae as long as they [meet the requirements](Versions.md).
+
+## Merging, rebasing, cherry-picking
+
+PRs modifying formulae that don't need bottles or making changes that don't
+require new bottles to be pulled should use GitHub's squash & merge or rebase & merge workflows.
+See the [table below](#how-to-merge-without-bottles) for more details.
+
+Otherwise, you should use `brew pr-pull` (or `rebase`/`cherry-pick` contributions).
+
+Don’t `rebase` until you finally `push`. Once `master` is pushed, you can’t
+`rebase`: **you’re a maintainer now!**
+
+Cherry-picking changes the date of the commit, which kind of sucks.
+
+Don’t `merge` unclean branches. So if someone is still learning `git` and
+their branch is filled with nonsensical merges, then `rebase` and squash
+the commits. Our main branch history should be useful to other people,
+not confusing.
 
-This is a guiding principle. As a maintainer, you can make a call to either
-request changes from a contributor or help them out based on their comfort and
-previous contributions. Remember, as a team we
-[Prioritise Maintainers Over Users](Maintainers-Avoiding-Burnout.md) to avoid
-burnout.
+Here’s a flowchart for managing a PR which is ready to merge:
 
-This is a more practical checklist; it should be used after you get familiar with
-[Maintainer Guidelines](Maintainer-Guidelines.md).
+![Flowchart for managing pull requests](assets/img/docs/managing-pull-requests.drawio.svg)
 
-## Checklist
+Only one maintainer is necessary to approve and merge the addition of a new or updated formula which passes CI. However, if the formula addition or update proves controversial the maintainer who adds it will be expected to answer requests and fix problems that arise with it in future.
 
-Check for:
+### How to merge without bottles
+
+Here are guidelines about when to use squash & merge versus rebase & merge. These options should only be used with PRs where bottles are not needed or affected.
```

Comment:
```suggestion
Here are guidelines about when to use squash & merge versus rebase & merge. These options should only be used with PRs where bottles are not affected.
```

Because 🔜 bottles will always be needed.